src/Simple/JSON.purs

```diff
@@ -226,10 +225,13 @@ exceptTApply :: forall a b e m. Semigroup e => Applicative m => ExceptT e m (a -
 exceptTApply fun a = ExceptT $ ado
   fun' :: Either e (a -> b) <- runExceptT fun
   a' :: (Either e a) <- runExceptT a
-  in toEither (apply (fromEither fun') (fromEither a'))
+  in applyEither fun' a'
 
-fromEither :: forall e a. Semigroup e => Either e a -> V e a
-fromEither = either invalid pure
+applyEither :: forall e a b. Semigroup e => Either e (a -> b) -> Either e a -> Either e b
+applyEither (Left e) (Right _) = Left e
+applyEither (Left e1) (Left e2) = Left (e1 <> e2)
+applyEither (Right _) (Left e) = Left e
+applyEither (Right fun) (Right a) = Right (fun a)
```

Comment:
yeah, i think this probably works fine, and it'll get quickly instanceof matched by javascript engines anyhow